src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -325,6 +325,175 @@ std::optional<Values> Visitor::visit(Parser::ValuesClauseContext* ctx) {
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::CopyContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Copy is not supported.");
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::InsertDataContext* ctx) {
+  return visit(ctx->quadData());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::DeleteDataContext* ctx) {
+  return visit(ctx->quadData());
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::DeleteWhereContext* ctx) {
+  // DeleteData only deletes the specified triples. DeleteWhere matches the
+  // given pattern and then deletes the matched triples. This is a shortcut for
+  // a modify query.
+  reportNotSupported(ctx, "SPARQL 1.1 Update DeleteWhere is not supported.");
+}
+
+// ____________________________________________________________________________________
+ParsedQuery Visitor::visit(Parser::ModifyContext* ctx) {
+  if (ctx->iri()) {
+    // Could also be default; disallow completely for now.
+    reportNotSupported(ctx->iri(), "Named graphs are not supported.");
+  }
+  if (!ctx->usingClause().empty()) {
+    reportNotSupported(ctx->usingClause(0),
+                       "Using is not supported in Update/Insert.");
+  }
+
+  parsedQuery_._rootGraphPattern = visit(ctx->groupGraphPattern());
+
+  parsedQuery_._clause = parsedQuery::UpdateClause();
+  if (ctx->deleteClause()) {
+    parsedQuery_.updateClause().toDelete_ = visit(ctx->deleteClause());
+  }
+  if (ctx->insertClause()) {
+    parsedQuery_.updateClause().toInsert_ = visit(ctx->insertClause());
+  }
+
+  return parsedQuery_;
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::DeleteClauseContext* ctx) {
+  return visit(ctx->quadPattern());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::InsertClauseContext* ctx) {
+  return visit(ctx->quadPattern());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::QuadPatternContext* ctx) {
+  return visit(ctx->quads());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::QuadDataContext* ctx) {
+  auto quads = visit(ctx->quads());
+  auto checkAndReportVar = [&ctx](const TripleComponent& term) {
+    if (term.isVariable()) {
+      reportError(ctx->quads(), "Variables (" + term.getVariable().name() +
+                                    ") are not allowed here.");
```

Comment:
Why is that?
Is that a semantic requirement of SPARQL, o is that just because you don't actually support the code for WHERE clauses in update?
In the latter case you shouldn't check this here, but finish the parser in this PR, and later check this for now in the parsedQuery (all the stuff you can parse, but not yet compute).

Reply:
That is  requirement of the Specs ([Note 8 on the Grammar in 19.8](https://www.w3.org/TR/sparql11-query/#sparqlGrammar))